src/dotty/tools/dotc/core/Types.scala

```diff
@@ -2358,7 +2374,7 @@ object Types {
     // no customized hashCode/equals needed because cycle is broken in PolyType
     override def toString = s"PolyParam(${binder.paramNames(paramNum)})"
 
-    override def computeHash = doHash(paramNum, binder)
+    override def computeHash = finishHash(hashing.mix(hashing.mix(hashSeed, paramNum), binder.identityHash), 1)
```

Comment:
For consistency, I'd either use one of the existing doHash methods or create a new one that fits. Hashmin.mix is not called directly in Types, and we should keep it that way.